torch/onnx/_internal/exporter.py

```diff
@@ -155,21 +156,241 @@ def serialize(
         destination.write(export_output.model_proto.SerializeToString())
 
 
+# TODO(bowbao): Add diagnostics for IO adapters.
+@runtime_checkable
+class InputAdaptStep(Protocol):
+    """A protocol that defines a step in the input adapting process.
+
+    The input adapting process is a sequence of steps that are applied to the
+    PyTorch model inputs to transform them into the inputs format expected by the
+    exported ONNX model. Each step takes the PyTorch model inputs as arguments and
+    returns the transformed inputs.
+
+    This serves as a base formalized construct for the transformation done to model
+    input signature by any individual component in the exporter.
+    """
+
+    def adapt(
+        self, model_args: Sequence[Any], model_kwargs: Mapping[str, Any]
+    ) -> Tuple[Sequence[Any], Mapping[str, Any]]:
+        ...
+
+
+class InputAdapter:
+    """A class that adapts the PyTorch model inputs to exported ONNX model inputs format."""
+
+    _input_adapt_steps: List[InputAdaptStep]
+
+    def __init__(self, input_adapt_steps: Optional[List[InputAdaptStep]] = None):
+        self._input_adapt_steps = input_adapt_steps or []
+
+    @_beartype.beartype
+    def append_step(self, step: InputAdaptStep) -> None:
+        """Appends a step to the input adapt steps.
+
+        Args:
+            step: The step to append.
+        """
+        self._input_adapt_steps.append(step)
+
+    @_beartype.beartype
+    def to_onnx(self, *model_args, **model_kwargs) -> Sequence[torch.Tensor]:
```

Comment:
```suggestion
    def apply(self, *model_args, **model_kwargs) -> Sequence[torch.Tensor]:
```
Evrry time that I read this, I expect to see a `ModelProto` lol. There is nothing in this class that restricts it to onnx, so we could use a generic term such as `apply`